### ISSUE TITLE
Cannot @InjectMock JsonWebToken since 2.8.0

### ISSUE THREAD

**Issue by pschyma** [quarkusio/quarkus#25095]
### Describe the bug

We are using `@InjectMock` (from https://github.com/quarkiverse/quarkus-mockk) to prepare the JWT for tests where we cannot use `@TestSecurity` or WireMock - e.g. in parameterized service tests where the token is depending on the test parameters. 

Since the upgrade to Quarkus 2.8.0.Final the injection fails with an Exception.

### Expected behavior

A mocked instance is injected into the test that can be used.

### Actual behavior

An exception is thrown:
```
Failed to create test instance
org.junit.jupiter.api.extension.TestInstantiationException: Failed to create test instance
	at io.quarkus.test.junit.QuarkusTestExtension.initTestState(QuarkusTestExtension.java:797)
	at io.quarkus.test.junit.QuarkusTestExtension.interceptTestClassConstructor(QuarkusTestExtension.java:751)
	at org.junit.jupiter.engine.execution.ExecutableInvoker.lambda$invoke$0(ExecutableInvoker.java:105)
	at org.junit.jupiter.engine.execution.InvocationInterceptorChain$InterceptedInvocation.proceed(InvocationInterceptorChain.java:106)
	at org.junit.jupiter.api.extension.InvocationInterceptor.interceptTestClassConstructor(InvocationInterceptor.java:73)
	at org.junit.jupiter.engine.execution.ExecutableInvoker.lambda$invoke$0(ExecutableInvoker.java:105)
	at org.junit.jupiter.engine.execution.InvocationInterceptorChain$InterceptedInvocation.proceed(InvocationInterceptorChain.java:106)
	at org.junit.jupiter.engine.execution.InvocationInterceptorChain.proceed(InvocationInterceptorChain.java:64)
	at org.junit.jupiter.engine.execution.InvocationInterceptorChain.chainAndInvoke(InvocationInterceptorChain.java:45)
	at org.junit.jupiter.engine.execution.InvocationInterceptorChain.invoke(InvocationInterceptorChain.java:37)
	at org.junit.jupiter.engine.execution.ExecutableInvoker.invoke(ExecutableInvoker.java:104)
	at org.junit.jupiter.engine.execution.ExecutableInvoker.invoke(ExecutableInvoker.java:77)
	at org.junit.jupiter.engine.descriptor.ClassBasedTestDescriptor.invokeTestClassConstructor(ClassBasedTestDescriptor.java:355)
	at org.junit.jupiter.engine.descriptor.ClassBasedTestDescriptor.instantiateTestClass(ClassBasedTestDescriptor.java:302)
	at org.junit.jupiter.engine.descriptor.ClassTestDescriptor.instantiateTestClass(ClassTestDescriptor.java:79)
	at org.junit.jupiter.engine.descriptor.ClassBasedTestDescriptor.instantiateAndPostProcessTestInstance(ClassBasedTestDescriptor.java:280)
	at org.junit.jupiter.engine.descriptor.ClassBasedTestDescriptor.lambda$testInstancesProvider$4(ClassBasedTestDescriptor.java:272)
	at java.base/java.util.Optional.orElseGet(Optional.java:369)
	at org.junit.jupiter.engine.descriptor.ClassBasedTestDescriptor.lambda$testInstancesProvider$5(ClassBasedTestDescriptor.java:271)
	at org.junit.jupiter.engine.execution.TestInstancesProvider.getTestInstances(TestInstancesProvider.java:31)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.lambda$prepare$0(TestMethodTestDescriptor.java:102)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.prepare(TestMethodTestDescriptor.java:101)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.prepare(TestMethodTestDescriptor.java:66)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$prepare$2(NodeTestTask.java:123)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.prepare(NodeTestTask.java:123)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:90)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1541)
	at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.invokeAll(SameThreadHierarchicalTestExecutorService.java:41)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$6(NodeTestTask.java:155)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:141)
	at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:137)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$9(NodeTestTask.java:139)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:138)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:95)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1541)
	at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.invokeAll(SameThreadHierarchicalTestExecutorService.java:41)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$6(NodeTestTask.java:155)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:141)
	at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:137)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$9(NodeTestTask.java:139)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:138)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:95)
	at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.submit(SameThreadHierarchicalTestExecutorService.java:35)
	at org.junit.platform.engine.support.hierarchical.HierarchicalTestExecutor.execute(HierarchicalTestExecutor.java:57)
	at org.junit.platform.engine.support.hierarchical.HierarchicalTestEngine.execute(HierarchicalTestEngine.java:54)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:107)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:88)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.lambda$execute$0(EngineExecutionOrchestrator.java:54)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.withInterceptedStreams(EngineExecutionOrchestrator.java:67)
	at org.junit.platform.launcher.core.EngineExecutionOrchestrator.execute(EngineExecutionOrchestrator.java:52)
	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:114)
	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:86)
	at org.junit.platform.launcher.core.DefaultLauncherSession$DelegatingLauncher.execute(DefaultLauncherSession.java:86)
	at org.junit.platform.launcher.core.SessionPerRequestLauncher.execute(SessionPerRequestLauncher.java:53)
	at org.gradle.api.internal.tasks.testing.junitplatform.JUnitPlatformTestClassProcessor$CollectAllTestClassesExecutor.processAllTestClasses(JUnitPlatformTestClassProcessor.java:99)
	at org.gradle.api.internal.tasks.testing.junitplatform.JUnitPlatformTestClassProcessor$CollectAllTestClassesExecutor.access$000(JUnitPlatformTestClassProcessor.java:79)
	at org.gradle.api.internal.tasks.testing.junitplatform.JUnitPlatformTestClassProcessor.stop(JUnitPlatformTestClassProcessor.java:75)
	at org.gradle.api.internal.tasks.testing.SuiteTestClassProcessor.stop(SuiteTestClassProcessor.java:61)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:36)
	at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:24)
	at org.gradle.internal.dispatch.ContextClassLoaderDispatch.dispatch(ContextClassLoaderDispatch.java:33)
	at org.gradle.internal.dispatch.ProxyDispatchAdapter$DispatchingInvocationHandler.invoke(ProxyDispatchAdapter.java:94)
	at com.sun.proxy.$Proxy5.stop(Unknown Source)
	at org.gradle.api.internal.tasks.testing.worker.TestWorker$3.run(TestWorker.java:193)
	at org.gradle.api.internal.tasks.testing.worker.TestWorker.executeAndMaintainThreadName(TestWorker.java:129)
	at org.gradle.api.internal.tasks.testing.worker.TestWorker.execute(TestWorker.java:100)
	at org.gradle.api.internal.tasks.testing.worker.TestWorker.execute(TestWorker.java:60)
	at org.gradle.process.internal.worker.child.ActionExecutionWorker.execute(ActionExecutionWorker.java:56)
	at org.gradle.process.internal.worker.child.SystemApplicationClassLoaderWorker.call(SystemApplicationClassLoaderWorker.java:133)
	at org.gradle.process.internal.worker.child.SystemApplicationClassLoaderWorker.call(SystemApplicationClassLoaderWorker.java:71)
	at worker.org.gradle.process.internal.worker.GradleWorkerMain.run(GradleWorkerMain.java:69)
	at worker.org.gradle.process.internal.worker.GradleWorkerMain.main(GradleWorkerMain.java:74)
Caused by: javax.enterprise.context.ContextNotActiveException
	at io.quarkus.arc.impl.ClientProxies.getDelegate(ClientProxies.java:46)
	at org.eclipse.microprofile.jwt.OidcJsonWebTokenProducer_ProducerMethod_currentAccessToken_6cc9acf87bb9cb50add3593d880ba325b4a9f587_ClientProxy.arc$delegate(Unknown Source)
	at org.eclipse.microprofile.jwt.OidcJsonWebTokenProducer_ProducerMethod_currentAccessToken_6cc9acf87bb9cb50add3593d880ba325b4a9f587_ClientProxy.hashCode(Unknown Source)
	at io.quarkiverse.test.junit.mockk.internal.MocksTracker$Mocked.hashCode(MocksTracker.kt)
	at java.base@11.0.12/java.util.HashMap.hash(HashMap.java:339)
	at java.base@11.0.12/java.util.HashMap.put(HashMap.java:607)
	at java.base@11.0.12/java.util.HashSet.add(HashSet.java:220)
	at io.quarkiverse.test.junit.mockk.internal.MocksTracker.track(MocksTracker.kt:11)
	at io.quarkiverse.test.junit.mockk.internal.CreateMockkMocksCallback.afterConstruct(CreateMockkMocksCallback.kt:21)
	at java.base@11.0.12/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base@11.0.12/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base@11.0.12/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base@11.0.12/java.lang.reflect.Method.invoke(Method.java:566)
	at app//io.quarkus.test.junit.QuarkusTestExtension.initTestState(QuarkusTestExtension.java:786)
	... 81 more
``` 

### How to Reproduce?

Simple reproducer:
```kotlin
import io.mockk.every
import io.quarkiverse.test.junit.mockk.InjectMock
import io.quarkus.test.junit.QuarkusTest
import org.assertj.core.api.Assertions.assertThat
import org.eclipse.microprofile.jwt.JsonWebToken
import org.junit.jupiter.api.Test

@QuarkusTest
class SimpleTest {
  @InjectMock lateinit var accessToken: JsonWebToken

  @Test
  fun `should inject token`() {
    every { accessToken.subject } returns "test"

    assertThat(accessToken.subject).isEqualTo("test")
  }
}
```

### Output of `java -version`

OpenJDK Runtime Environment Temurin-11.0.14+9 (build 11.0.14+9)

### Quarkus version or git rev

2.8.{0,1}.Final

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

Gradle 7.4.1

**Comment by sberyozkin** [comment]
@pschyma Thanks, I don't recall anything changing in the related OIDC token producer code since `2.7.5`. 
@geoand Perhaps something has changed in the test support code ? @mkouba May be something has changed in the way requests contexts are managed ? I believe it works in 2.7.5

**Comment by geoand** [comment]
I am not aware of any relevant change

**Comment by sberyozkin** [comment]
@glefloch Hi, maybe `quarkus-mockk` needs to be updated somehow alongside the update to `2.8.0` ?

**Comment by mkouba** [comment]
> May be something has changed in the way requests contexts are managed ?

I have no idea.  We need to know _when exactly_ is the `OidcJsonWebTokenProducer` invoked to find out why the request context is not active.

**Comment by glefloch** [comment]
I released a `1.1.0` version of the library, there was an API change in quarkus 2.8.0, could you try with it?

**Comment by sberyozkin** [comment]
@pschyma See @glefloch's comment, can you try `1.1.0` version of `quarkus.mockk` ?

**Comment by sberyozkin** [comment]
@mkouba

>We need to know when exactly is the OidcJsonWebTokenProducer invoked to find out why the request context is not active.

It is invoked whenever `@Inject JsonWebToken jwt;` code is present, I'm not sure what you mean by `when exactly` - is it something that the request context injection code controls ?

**Comment by mkouba** [comment]
> I'm not sure what you mean by when exactly - is it something that the request context injection code controls ?

Well, what I mean is that the request context must be active, for example the request context is active when processing an HTTP request. However, in this particular case it seems that the `OidcJsonWebTokenProducer` should not be used and instead a `JsonWebToken` mock should be injected, or?

**Comment by pschyma** [comment]
@sberyozkin @glefloch I'm using it already as the older version does not work with Quarkus 2.8.0.

Actually, I was not sure where to raise this issue, here or against quarkus-mockk. But since my original issue was resolved at quarkus-mockk and `@InjectMock` still works for services, I've ended here.

**Comment by sberyozkin** [comment]
@mkouba I see. With [@OidcSecurity](https://github.com/quarkusio/quarkus/blob/main/integration-tests/oidc-tenancy/src/test/java/io/quarkus/it/keycloak/TestSecurityLazyAuthTest.java#L27) used alongside `@TestSecurity`, a test `JsonWebToken` generated indirectly in the `TestIdentityProvider` there `OidcJsonWebTokenProducer` is invoked.
I'm not sure how `OidcJsonWebTokenProducer` can be skipped in this case, I guess `quarkus-mockk` should provide a higher priority/alternative producer. 
@glefloch, @pschyma  do you know how it works with 2.7.5, specifically why `OidcJsonWebTokenProducer` is not invoked ?

**Comment by glefloch** [comment]
@sberyozkin nothing changed between `2.7.x` and `2.8.x`, the `quarkus-junit5-mockk` works as the `quarkus-junit5-mockito` extension. If we need to fix something, it should be done on both library. @pschyma do you have a reproducer?

**Comment by glefloch** [comment]
Looking at the stacktrace, it looks like, the error is thrown right after the mock is created. 

Mocks are tracked in a map that contains both the mock and the bean instance (queried using `Arc.container().instance(...)`) 
When adding that to the map, it triggers the `hashcode` method of the bean instance, which then throw that exception.

**Comment by pschyma** [comment]
I've attached a reproducer. 

[oidc-reproducer.zip](https://github.com/quarkusio/quarkus/files/8569996/oidc-reproducer.zip)

**Comment by glefloch** [comment]
I have the issue. It comes from the `quarkus-mockk` I will a fix today and release a new version of the library too

**Comment by glefloch** [comment]
@pschyma I released `quarkus-junit5-mockk:1.1.1`, it be indexed soon

**Comment by sberyozkin** [comment]
@glefloch Nice, thanks for this fix
